nexus/src/db/datastore/organization.rs

```diff
@@ -51,27 +54,54 @@ impl DataStore {
         let organization = Organization::new(organization.clone(), silo_id);
         let name = organization.name().as_str().to_string();
 
-        Silo::insert_resource(
-            silo_id,
-            diesel::insert_into(dsl::organization).values(organization),
-        )
-        .insert_and_get_result_async(self.pool_authorized(opctx).await?)
-        .await
-        .map_err(|e| match e {
-            AsyncInsertError::CollectionNotFound => Error::InternalError {
-                internal_message: format!(
-                    "attempting to create an \
-                    organization under non-existent silo {}",
-                    silo_id
-                ),
-            },
-            AsyncInsertError::DatabaseError(e) => {
-                public_error_from_diesel_pool(
-                    e,
-                    ErrorHandler::Conflict(ResourceType::Organization, &name),
+        self.pool_authorized(opctx)
+            .await?
+            .transaction_async(|conn| async move {
+                let org = Silo::insert_resource(
+                    silo_id,
+                    diesel::insert_into(dsl::organization).values(organization),
                 )
-            }
-        })
+                .insert_and_get_result_async(&conn)
+                .await
+                .map_err(|e| match e {
+                    AsyncInsertError::CollectionNotFound => {
+                        Error::InternalError {
```

Comment:
I think this matches the previous behavior (so not a blocker), but I'd expect this to be a 404.

Reply:
Done in 4e6ea232bb8424d1f5666cdbe779b02ac596793b